src/hooks/use-last-seen-version.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect, useCallback } from 'react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+interface UseLastSeenVersion {
+	lastSeenVersion: string | undefined;
+	isNewVersion: boolean;
+	updateLastSeenVersion: () => Promise< void >;
+	isLoading: boolean;
+}
+
+export function useLastSeenVersion(): UseLastSeenVersion {
+	const [ lastSeenVersion, setLastSeenVersion ] = useState< string | undefined >( undefined );
+	const [ isLoading, setIsLoading ] = useState( true );
+	const currentVersion = window.appGlobals.appVersion;
+
+	useEffect( () => {
+		const fetchLastSeenVersion = async () => {
+			try {
+				const version = await getIpcApi().getLastSeenVersion();
+				setLastSeenVersion( version );
+			} catch ( error ) {
+				console.error( 'Failed to get last seen version:', error );
+			} finally {
+				setIsLoading( false );
+			}
+		};
+		fetchLastSeenVersion();
+	}, [] );
+
+	const updateLastSeenVersion = useCallback( async () => {
+		try {
+			await getIpcApi().saveLastSeenVersion( currentVersion );
+			setLastSeenVersion( currentVersion );
+		} catch ( error ) {
+			console.error( 'Failed to save last seen version:', error );
+		}
+	}, [ currentVersion ] );
+
+	const isNewVersion = ! isLoading && !! currentVersion && lastSeenVersion !== currentVersion;
```

Comment:
For simplicity this PR displays the "What's New" modal when the current version of the Studio app is _different_ from the `lastSeenVersion`.

We will adjust this logic in the subsequent PR where the modal will be triggered only when Major or Minor versions are changed - or - when a hardcoded `forceWhatsNew` will be set to `true`.